pkg/pillar/cmd/upgradeconverter/movepatchenvelopes.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package upgradeconverter
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+const (
+	srcDir  = "/zedrouter/AppInstMetaData"
+	destDir = "/msrv/AppInstMetaData"
+)
+
+func exists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true, nil
+	}
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+func moveFile(sourcePath, destPath string) error {
+	inputFile, err := os.Open(sourcePath)
+	if err != nil {
+		return fmt.Errorf("Couldn't open source file: %v", err)
+	}
+	defer inputFile.Close()
+
+	outputFile, err := os.Create(destPath)
+	if err != nil {
+		return fmt.Errorf("Couldn't open dest file: %v", err)
+	}
+	defer outputFile.Close()
+
+	_, err = io.Copy(outputFile, inputFile)
+	if err != nil {
+		return fmt.Errorf("Couldn't copy to dest from source: %v", err)
+	}
+
+	inputFile.Close() // https://stackoverflow.com/a/64943554/246801
+
+	err = os.Remove(sourcePath)
+	if err != nil {
+		return fmt.Errorf("Couldn't remove source file: %v", err)
+	}
+	return nil
+}
+
+// moveDir copies a whole directory recursively
+func moveDir(src string, dst string) error {
+	var err error
+	var fds []os.FileInfo
+	var srcinfo os.FileInfo
+
+	if srcinfo, err = os.Stat(src); err != nil {
+		return err
+	}
+
+	if err = os.MkdirAll(dst, srcinfo.Mode()); err != nil {
+		return err
+	}
+
+	if fds, err = ioutil.ReadDir(src); err != nil {
+		return err
+	}
+	for _, fd := range fds {
+		srcfp := path.Join(src, fd.Name())
+		dstfp := path.Join(dst, fd.Name())
+
+		if fd.IsDir() {
+			if err = moveDir(srcfp, dstfp); err != nil {
+				fmt.Println(err)
+			}
+		} else {
+			if err = moveFile(srcfp, dstfp); err != nil {
+				fmt.Println(err)
+			}
+		}
+	}
+	return nil
+}
+
+func movePersistPubsub(ctxPtr *ucContext) error {
+	src := filepath.Join(ctxPtr.persistStatusDir, srcDir)
+	dst := filepath.Join(ctxPtr.persistStatusDir, destDir)
+
+	srcExists, err := exists(src)
+	if err != nil {
+		return fmt.Errorf("stat zedrouter failed %v", err)
+	}
+
+	dstExists, err := exists(dst)
+	if err != nil {
+		return fmt.Errorf("stat msrv failed %v", err)
+	}
+
+	if srcExists {
+		if dstExists {
+			// if there is already a folder we are not going to
+			// override it
+			return os.RemoveAll(src)
+		} else {
+			if err := moveDir(src, dst); err != nil {
```

Comment:
How about using `fileutils.CopyDir`?

Reply:
I mean this: https://github.com/lf-edge/eve/blob/master/pkg/pillar/utils/file/copy.go#L80